src/main/java/io/cdap/plugin/gcp/publisher/source/PubSubRDDIterator.java

```diff
@@ -150,15 +151,21 @@ private List<ReceivedMessage> fetchAndAck(long timeout) throws IOException {
 
     // From  the docs - "A pull response that comes with 0 messages must not be used as an indicator that there are no
     // messages in the backlog. It's possible to get a response with 0 messages and have a subsequent
-    // request that returns messages." Since 0 messages do not indicate a lack of available messages, no delays are
-    // included  between retries.
+    // request that returns messages." Since 0 messages do not indicate a lack of available messages, a fixed delay
+    // is included  between retries instead of exponential backoff.
     // Messages will be read till batch duration completes, to account for messages published
     // later in the batch. No documented quota on the number of synchronous subscriber requests and no issues observed
     // in testing when repeated calls are made when no data is available in the subscription.
-    while (System.currentTimeMillis() < timeout) {
+    while (System.currentTimeMillis() < expiryTimeMillis) {
       PullResponse pullResponse = subscriber.pullCallable().call(pullRequest);
       List<ReceivedMessage> receivedMessagesList = pullResponse.getReceivedMessagesList();
       if (receivedMessagesList.isEmpty()) {
+        // Delay before re-attempt
+        try {
+          TimeUnit.MILLISECONDS.sleep(RETRY_DELAY);
+        } catch (InterruptedException e) {
+          LOG.debug("Interrupted while waiting for retry. ", e);
```

Comment:
if interrupted, we should also return an empty list so that the iterator will stop.

Reply:
Done.